model-optimizer/extensions/middle/MarkSubgraphsWithCorrectLayout.py

```diff
@@ -172,6 +202,45 @@ def walk_up_from_in_ports_to_out_ports(in_ports: Set[Port], out_ports: Set[Port]
             visited_nodes.add(source_node)
         return visited_ports, visited_nodes
 
+    @staticmethod
+    def walk_up_from_in_ports_to_out_ports_not_going_into_conv_weights(in_ports: Set[Port], out_ports: Set[Port],
+                                                                       visited_ports: Set[Port] = None,
+                                                                       visited_nodes: Set[Node] = None):
+        get_weights_port_index = lambda node: node.weights_index if node.has_valid('weights_index') else 1
+        weighted_layer_type_to_in_weights_port = {
+            'Convolution': get_weights_port_index,
```

Comment:
What about `ConvolutionBackpropData` and `GroupConvolutionBackpropData`?

Reply:
They appear later on the back phase